File: python/src/wallets/crossmint/goat_wallets/crossmint/mint_plugin.py
from typing import Callable
from goat import Chain, PluginBase
from .mint import CrossmintMintService
from .api_client import CrossmintWalletsAPI
from .chains import is_chain_supported_by_minting

class MintPlugin(PluginBase):
    def __init__(self, client: CrossmintWalletsAPI):
        super().__init__("mint", [CrossmintMintService(client)])

    def supports_chain(self, chain: Chain) -> bool:
        if chain["type"] == "evm":
            return is_chain_supported_by_minting(chain["id"] or 0)
        return chain["type"] in ["aptos", "solana"]

def mint_plugin(client: CrossmintWalletsAPI) -> Callable[[], MintPlugin]:
    return lambda: MintPlugin(client)

File: typescript/packages/wallets/crossmint/src/plugins/mint.plugin.ts
import type { CrossmintApiClient } from "@crossmint/common-sdk-base";
import { type Chain, PluginBase } from "@goat-sdk/core";
import type { EVMWalletClient } from "@goat-sdk/wallet-evm";
import { isChainSupportedByMinting } from "../chains";
import { CrossmintMintService } from "./mint.service";

export class MintPlugin extends PluginBase<EVMWalletClient> {
    constructor(client: CrossmintApiClient) {
        super("mint", [new CrossmintMintService(client)]);
    }

    supportsChain(chain: Chain) {
        if (chain.type === "evm") {
            return isChainSupportedByMinting(chain.id ?? 0);
        }

        if (chain.type === "aptos" || chain.type === "solana") {
            return true;
        }

        return false;
    }
}

export function mintPlugin(client: CrossmintApiClient) {
    return () => {
        return new MintPlugin(client);
    };
}
